AppKit/CPStringDrawing.j

```diff
@@ -74,6 +75,12 @@ CPStringSizeCachingEnabled = YES;
         CPStringSizeMeasuringContext.font = [aFont cssString];
         var teststring = "0123456879abcdefghiklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ,.-()";
         CPStringSizeIsCanvasSizingInvalid = ABS([CPPlatformString sizeOfString:teststring withFont:aFont forWidth:0].width - CPStringSizeMeasuringContext.measureText(teststring).width) > 2;
+
+        CPStringSizeNeedsAlwaysSetFont = NO;
+        var ua = navigator.userAgent.toLowerCase();
+        
+        if (ua.indexOf('safari') !== -1 && ua.indexOf('chrome') === -1)
+            CPStringSizeNeedsAlwaysSetFont = YES; // Safari
```

Comment:
We want you to check things like this in ```CPCompatibility.j```. Add a feature (please document it well) and use the provided engine checks.

Ask if the feature is present from this method.